opencdc/record.go

```diff
@@ -147,3 +148,9 @@ func (r Record) Clone() Record {
 	}
 	return clone
 }
+
+func (r Record) AttachSchema(s schema.Instance) {
+	r.Metadata.SetSchemaType(schema.TypeAvro)
+	r.Metadata.SetSchemaName(s.Name)
+	r.Metadata.SetSchemaVersion(s.Version)
+}
```

Comment:
I'd rather turn the dependency around and have utility functions regarding schemas in the `schema` package. We'll probably need more, e.g. for decoding/encoding a record.

Reply:
For decoding and encoding, I think that makes sense. For attaching the schema, I find it quite natural to write something like `record.AttachSchema(s)`.

Reply:
I'm not opposed to the method itself, the issue is organizing the code without circular dependencies. Once we create a dependency from `opencdc` -> `schema` we can't have it the other way around anymore. And I'd rather keep the `opencdc` package lean and focused on records, and stuff anything regarding to schemas into `schema`, that includes utilities to work with records, encoding, decoding, etc.

Reply:
I see... But why not have those utilities in `opencdc`? Because, it's records that are encoded, so having all functions where records are the "target" in one package -- opencdc -- makes sense to me. I'd expect only schema-related functions in the `schema` package, and from that standpoint, I wouldn't expect the `schema` package to reference records (or anything OpenCDC).

Reply:
As I said, my concern are circular dependencies. The `opencdc` package is used by any package that works with records, that's like half of the packages in Conduit. If we're not careful we might quickly get a circular dependency, (e.g. in the future we might want to import package `X` in `schema`, but `X` imports `opencdc` and we're screwed). I'd rather see that `opencdc` is not importing any other package (from our codebase) without a good reason because it has such a central role.

Reply:
Gotcha, I thought you meant a circular dependency between `opencdc` and `schema`, but you actually meant circular dependencies when using those packages in Conduit for example. That's a valid concern. :+1: